src/frontend/src/lib/utils/nav.utils.ts

```diff
@@ -48,6 +48,25 @@ const tokenUrl = ({
 export const networkParam = (networkId: NetworkId | undefined): string =>
 	isNullish(networkId) ? '' : `${NETWORK_PARAM}=${networkId.description ?? ''}`;
 
+export const networkUrl = ({
+	path,
+	networkId,
+	isTransactionsRoute,
+	fromRoute
+}: {
+	path: AppPath;
+	networkId: Option<NetworkId>;
+	isTransactionsRoute: boolean;
+	fromRoute: NavigationTarget | null;
+}) =>
+	isTransactionsRoute
+		? nonNullish(fromRoute?.url.searchParams.get(NETWORK_PARAM))
```

Comment:
`notEmptyString` instead of `nonNullish` maybe? not sure if that would be an issue to have an empty param but, the url could be a bit more cute?